pyi.Dockerfile

```
@@ -290,7 +293,7 @@ COPY ./scripts/list_needed_libs.sh ./scripts/list_needed_libs.sh
 # using scl here to get the proper LD_LIBRARY_PATH set
 # hadolint ignore=SC2046,SC2086
 RUN set -e; \
-    if [ $(uname -m) != "aarch64" ]; then \
+    if [ $(uname -m) != "aarch64" ] && [ "$STATICX" = "true" ]; then \
```

```suggestion
    if [ "$STATICX" = "true" ]; then \
```
`staticx` should be run on aarch64 for gprofiler to ensure correct loading.

---

Yup, it should :( we broke it. It is fixed in https://github.com/Granulate/gprofiler/pull/806